### PR TITLE
Automatic fetching of neural networks from CCDB for the TPC PID Response correction

### PR DESCRIPTION
**Pull request by ChSonnabend**
The neural networks will be used for the TPC PID response correction. Automatic fetching is necessary to run jobs over multiple run numbers etc. and has to be done on the fly on Hyperloop. This PR brings that functionality.

### CODE REVIEW

**Comment by ChSonnabend** [comment]
Whenever you have time, you can have a look @njacazio . This now fixes the behaviour for automatic fetching across different run number.

**Review by jgrosseo — CHANGES_REQUESTED** [review]
Sorry, this is not allowed... Those objects have to go to CCDB. We do not allow direct alien accesses as this can lead to overloads if there are not sufficient replicas...

**Comment by ChSonnabend** [comment]
But can the .onnx files be loaded into an object from CCDB?

**Comment by njacazio** [comment]
Hi @jgrosseo first of all, of course we are aware and we agree with you. However I think that we cannot store now .onnx files as this is not supported yet on CCDB. What do you suggest then?

**Comment by jgrosseo** [comment]
This cannot even be a c++ class, right? Hadn't we discussed in the past, I don't mean with the two of you but to solve this use case, to add a CCDB API function `CopyFile` which copies a file from a CCDB location to a local file and then it can be used? It may be that no one implemented this up to now, though. @sawenzel do you remember?
In principle CCDB is not bound to ROOT or C++ streamable objects I think. One can store a binary blob.

**Comment by njacazio** [comment]
> This cannot even be a c++ class, right? Hadn't we discussed in the past, I don't mean with the two of you but to solve this use case, to add a CCDB API function `CopyFile` which copies a file from a CCDB location to a local file and then it can be used? It may be that no one implemented this up to now, though. @sawenzel do you remember?
> In principle CCDB is not bound to ROOT or C++ streamable objects I think. One can store a binary blob.

Indeed the storing part is without issues, the issues are when retrieving the object.
As you say we could download the blob and then open it, let's see what Sandro says

**Comment by njacazio** [comment]
Hi, thanks a lot for your suggestions @jgrosseo ! Indeed with the CCDB API the blob can be retrieved and loaded in the workflow from a local file, this way we avoid both alien and CVMFS (that was another suggested way to go). I did a preliminary test and it seems to work, I let @ChSonnabend comment on the feasibility! Thanks again!

**Comment by jgrosseo** [comment]
So it already works with the existing CCDB API? I was about to make a JIRA ticket for the needed functionality in the API, but if you say it works, I wait a bit...

File: Common/Core/CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,7 @@ o2physics_target_root_dictionary(AnalysisCore
                         PID/DetectorResponse.h
                         PID/PIDTOF.h
                         PID/TOFReso.h
+                        PID/TPCPIDResponse.h
```
Comment (jgrosseo) [inline Common/Core/CMakeLists.txt:26]: I think you need to rebase...

**Comment by jgrosseo** [comment]
OK, great! I think you should also change the title then...

**Comment by ChSonnabend** [comment]
Hi @jgrosseo and @njacazio,
I think now it should be done, a simple fetch & merge from master was enough.

**Review by jgrosseo — APPROVED** [review]

**Comment by vkucera** [comment]
Merging this broke the macOS build for all subsequent PRs. @ChSonnabend Please fix it.

**Comment by ChSonnabend** [comment]
> Merging this broke the macOS build for all subsequent PRs. @ChSonnabend Please fix it.

It is fixed already with the latest bug-fix PR I did

**Comment by vkucera** [comment]
> > Merging this broke the macOS build for all subsequent PRs. @ChSonnabend Please fix it.
> 
> It is fixed already with the latest bug-fix PR I did

Thanks

**Pull request merged**